### PR TITLE
Propagate customData in FirebaseError when the user is disabled. 

### PR DESCRIPTION
**Pull request by prameshj**
Internal bug - b/229774279

### CODE REVIEW

File: packages/auth/demo/src/index.js [hunk 1]
```diff
@@ -277,7 +277,7 @@ function onAuthError(error) {
     // Handle second factor sign-in.
     handleMultiFactorSignIn(getMultiFactorResolver(auth, error));
   } else {
-    alertError('Error: ' + error.code);
+    alertError('Error: ' + error.code + 'user email : ' + error.customData.email);
```
Comment (sam-gc) [inline packages/auth/demo/src/index.js:280]: nit, put a space in front of "user email" so that it doesn't print right next to the error code

Comment (lisajian) [inline packages/auth/demo/src/index.js:280]: Not every error will have a customData.email field - maybe we just leave it as the error code for now?

Comment (prameshj) [reply packages/auth/demo/src/index.js:280]: I added this, so we can identify other error paths where the email is missing but should be propagated, during testing with the demo app.

File: packages/auth/src/api/authentication/idp.test.ts [hunk 2]
```diff
@@ -87,4 +87,35 @@ describe('api/authentication/signInWithIdp', () => {
...
+    let error: FirebaseError | null = null;
+    try {
+    await signInWithIdp(auth, request);
+    } catch (e) {
+      error = e;
+    }
+    expect(error!.customData!.email).to.eq("test123@gmail.com");
+    expect(error!.code).to.eq(
+      'auth/user-disabled'
+    );
```
Comment (sam-gc) [inline packages/auth/src/api/authentication/idp.test.ts:117]: You can use chai-as-promised to put all of this error testing logic into one assertion phrase. From ap/index.test.ts for example:
https://github.com/firebase/firebase-js-sdk/blob/b6196d28ce6117e17dbe87cfd554d568012ef31d/packages/auth/src/api/index.test.ts#L178-L183

i.e.

```js
await expect(signInWithIdp(auth, request)).to.be.rejectedWith('auth/user-disabled')
    .eventually.with.deep.property('customData', {email: 'test123@gmail.com'});
```

File: packages/auth/demo/src/index.js (hunk 1 above)
Comment (lisajian) [reply packages/auth/demo/src/index.js:280]: Right, what I mean is not every user that is signed up with Firebase Auth will have an email attached to them. For finding error paths where the email is missing, maybe we can leave that as a local change while we're developing

**Review by lisajian — COMMENTED** [review]
Could you attach a changeset and add an internal bug number or link to a Github issue to the PR description? Overall LGTM pending nits

**Review by egilmorez — APPROVED** [review]
Text string LG, thanks!

File: packages/auth/demo/src/index.js (hunk 1 above)
Comment (prameshj) [reply packages/auth/demo/src/index.js]: Sounds good, done.

File: packages/auth/src/api/authentication/idp.test.ts (hunk 2 above)
Comment (prameshj) [reply packages/auth/src/api/authentication/idp.test.ts]: Done.

File: packages/auth/demo/src/index.js (hunk 1 above)
Comment (prameshj) [reply packages/auth/demo/src/index.js]: Removed this line.

**Comment by prameshj** [comment]
> Could you attach a changeset and add an internal bug number or link to a Github issue to the PR description? Overall LGTM pending nits

Thanks! I am having trouble reproing the fix in the demo app now. I will submit the changeset once I resolve that.

**Review by sam-gc — APPROVED** [review]

**Comment by lisajian** [comment]
> 

SG, thanks! Giving a preemptive LGTM in the meantime. Thanks for the fix!

> > Could you attach a changeset and add an internal bug number or link to a Github issue to the PR description? Overall LGTM pending nits
> 
> Thanks! I am having trouble reproing the fix in the demo app now. I will submit the changeset once I resolve that.

**Review by lisajian — APPROVED** [review]

**Pull request merged**